tree.go

```diff
@@ -537,13 +536,13 @@ func (n *InternalNode) Get(k []byte, getter NodeResolverFn) ([]byte, error) {
 			return nil, errReadFromInvalid
 		}
 
-		payload, err := getter(child.commitment)
+		payload, err := getter(child.BytesCompressed())
 		if err != nil {
 			return nil, err
 		}
 
 		// deserialize the payload and set it as the child
-		c, err := ParseNode(payload, n.depth+1, child.commitment)
+		c, err := ParseNode(payload, n.depth+1, child.BytesCompressed())
 		if err != nil {
 			return nil, err
 		}
```

Comment:
All the **above** changes in this file are using the new API of `HashedNode`. Nothing new in the mechanics, just avoiding private fields and being clear about what we're getting.